tests/src/test/320+-noncdh-nondb/scala/org/apache/spark/sql/rapids/RapidsShuffleThreadedWriterSuite.scala

```diff
@@ -239,10 +240,16 @@ class RapidsShuffleThreadedWriterSuite extends FunSuite
 
   override def afterEach(): Unit = {
     TaskContext.unset()
+    blockIdToFileMap.clear()
+    temporaryFilesCreated.clear()
     try {
       Utils.deleteRecursively(tempDir)
-      blockIdToFileMap.clear()
-      temporaryFilesCreated.clear()
+    } catch {
+      case t: Throwable =>
```

Comment:
consider limiting to `IOException` or `NonFatal(t)`

Reply:
@gerashegalov handled here: https://github.com/NVIDIA/spark-rapids/pull/6640/commits/b654e6dfbdf97478c920eb358fcddc2c80cc7de2